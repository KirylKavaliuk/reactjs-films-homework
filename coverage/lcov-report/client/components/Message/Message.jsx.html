<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/Message/Message.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/Message</a> Message.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
&nbsp;
import withConditionalRendering from 'utils/rendering';
&nbsp;
import classNames from 'classnames';
&nbsp;
import styles from './Message.scss';
&nbsp;
class Message extends Component {
  timeout = null;
&nbsp;
  componentDidMount() {
    const { closeMessage } = this.props;
&nbsp;
    clearTimeout(this.timeout);
    this.timeout = setTimeout(() =&gt; closeMessage(), 4000);
  }
&nbsp;
  componentWillUnmount() {
    clearTimeout(this.timeout);
  }
&nbsp;
  render() {
    const { text, open, closeMessage } = this.props;
    const messageClasses = classNames(styles.message, { [styles.open]: open });
&nbsp;
    return (
      &lt;div className={ messageClasses } onClick={ closeMessage }&gt;
        { text }
      &lt;/div&gt;
    );
  }
}
&nbsp;
Message.defaultProps = {
  open: false,
  text: '',
};
&nbsp;
Message.propTypes = {
  open: PropTypes.bool,
  text: PropTypes.string,
};
&nbsp;
export default withConditionalRendering(Message, 'text');
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
