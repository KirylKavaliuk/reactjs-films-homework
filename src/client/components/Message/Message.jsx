import React, { Component } from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import styles from './Message.scss';

export default class Message extends Component {
  timeout = null;

  componentWillReceiveProps = (newProps) => {
    if (newProps.message.open) {
      clearTimeout(this.timeout);
      this.timeout = setTimeout(() => this.props.closeMessage(), 4000);
    }
  }

  render() {
    const { text, open } = this.props.message;

    return (
      <div
        className={
          classNames(
            styles.message,
            { [styles.open]: open },
          )
        }
        onClick={ this.props.closeMessage }
      >
        { text }
      </div>
    );
  }
}

Message.defaultProps = {
  open: false,
  text: '',
};

Message.propTypes = {
  open: PropTypes.bool,
  text: PropTypes.string,
};
