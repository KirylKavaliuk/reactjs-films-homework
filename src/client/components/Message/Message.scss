@import 'styles/palette';

.message {
  position: fixed;
  z-index: 101;
  bottom: -200px;
  left: 20px;
  box-sizing: border-box;
  width: 250px;
  padding: 15px;
  transition: all 0.45s ease;
  color: $WHITE;
  border-radius: 3px;
  background: $LIGHT_BLUE;
  font-size: 14px;
  line-height: 1.4;

  &:hover {
    cursor: pointer;
  }
}

.open {
  bottom: 20px;
}
