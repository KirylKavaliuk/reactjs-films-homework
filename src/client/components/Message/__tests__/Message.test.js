import React from 'react';
import renderer from 'react-test-renderer';
import Message from '../Message';

jest.useFakeTimers();

it('<Message/>', () => {
  const tree = renderer.create(<Message/>).toJSON();

  expect(tree).toBeNull();
  expect(tree).toMatchSnapshot();
});

it('<Message/>', () => {
  let open = true;

  const closeMessage = () => {
    open = false;
  };

  const tree = renderer.create(
    <Message
      open={ open }
      text='message'
      closeMessage={ closeMessage }
    />,
  );

  const message = tree.toJSON();

  jest.runAllTimers();

  expect(message).not.toBeNull();
  expect(message).toMatchSnapshot();

  tree.unmount();
});
