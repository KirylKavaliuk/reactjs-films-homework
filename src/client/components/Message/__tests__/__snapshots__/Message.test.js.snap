// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message/> 1`] = `null`;

exports[`<Message/> 2`] = `
<div
  className="Message__message___GEGYp undefined"
  onClick={[Function]}
>
  message
</div>
`;
